modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -542,3 +569,179 @@ class ApiClient(object):
+        :return: A tuple of (digest, prefix).
+          The digest is a hashing object that contains the cryptographic digest of the HTTP request.
+          The prefix is a string that identifies the cryptographc hash. It is used to generate the
+          'Digest' header as specified in RFC 3230.
+        """
+        if self.configuration.signing_scheme in ["rsa-sha512", "hs2019"]:
+            digest = SHA512.new()
+            prefix = "SHA-512="
+        elif self.configuration.signing_scheme in ["rsa-sha256"]:
+            digest = SHA256.new()
+            prefix = "SHA-256="
+        else:
+            raise Exception(
+                "Unsupported signing algorithm: {0}".format(self.configuration.signing_scheme))
+        digest.update(data)
+        return digest, prefix
+
+    def sign_digest(self, digest):
+        """
+        Signs a message digest with a private key specified in the configuration.
+
+        :param digest: A hashing object that contains the cryptographic digest of the HTTP request.
+        :return: A base-64 string representing the cryptographic signature of the input digest.
+        """
+        self.configuration.load_private_key()
+        privkey = self.configuration.private_key
+        if isinstance(privkey, RSA.RsaKey):
+            if self.configuration.signing_algorithm == 'PSS':
+                # RSASSA-PSS in Section 8.1 of RFC8017.
+                signature = pss.new(privkey).sign(digest)
+            elif self.configuration.signing_algorithm == 'PKCS1v15':
+                # RSASSA-PKCS1-v1_5 in Section 8.2 of RFC8017.
+                signature = PKCS1_v1_5.new(privkey).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(self.configuration.signing_algorithm))
+        elif isinstance(privkey, ECC.EccKey):
+            if self.configuration.signing_algorithm in ECDSA_KEY_SIGNING_ALGORITHMS:
+                signature = DSS.new(privkey, self.configuration.signing_algorithm).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(self.configuration.signing_algorithm))
+        else:
+            raise Exception("Unsupported private key: {0}".format(type(privkey)))
+        return b64encode(signature)
+
+    def get_authorization_header(self, signed_headers, signed_msg):
+        """
+        Calculates and returns the value of the 'Authorization' header when signing HTTP requests.
+        
+        :param signed_headers : A list of strings. Each value is the name of a HTTP header that
+           must be included in the HTTP signature calculation.
+        :param signed_msg: A base-64 encoded string representation of the signature.
+        :return: The string value of the 'Authorization' header, representing the signature
+           of the HTTP request.
+        """
+
+        headers_value = ""
+        is_created_set = False
+        for i, key in enumerate(signed_headers)
```

Comment:
Optionally how about changing this for loop into:
```
lower_keys = [k.lower() for k in signed_headers]
headers_value = " ".join(lower_keys)
```
That will save us two lines

Reply:
done.